### PR TITLE
Add toOptionalValidator

### PR DESCRIPTION
**Pull request by zoontek**
- Not sure about the wording. Maybe `toOptionalValidator`? (same goes for `emptyValue`)
- Should the empty value arg be required? Should the method only make string based validator optionals?

### CODE REVIEW

File: src/index.ts [hunk 1]
```diff
@@ -122,6 +122,28 @@ const isPromise = <T>(value: unknown): value is Promise<T> =>
...
   typeof (value as { then?: Function }).then === "function";
 
+export const createOptionalValidator =
+  <Value, ErrorMessage = string>(
+    validator: Validator<Value, ErrorMessage>,
+    emptyValue?: Value,
+  ): Validator<Value, ErrorMessage> =>
+  (value) => {
+    const isEmptyValue = value === (typeof emptyValue !== "undefined" ? emptyValue : "");
+    const result = validator(value);
```
Comment (bloodyowl) [inline src/index.ts:132]: maybe let's not run it in case `isEmptyValue` is true, could have side-effects or expensive computations

Comment (zoontek) [reply src/index.ts]: Ah yes, I'm stupid. Could be much more simpler.

File: src/index.ts [hunk 2]
```diff
@@ -153,6 +153,20 @@ export const combineValidators =
     }
   };
 
+export const toOptionalValidator =
+  <Value, ErrorMessage = string, EmptyValue extends Value = Value>(
```
Comment (bloodyowl) [inline src/index.ts:157]: can we rather get an `isEmptyValue: (x: Value) => boolean`? that'd help for values without referential equality.

Comment (zoontek) [reply src/index.ts]: Done https://github.com/swan-io/react-ux-form/pull/25/commits/ab1c5dc5e0d4874f9d6f3a2f9c8a396a9fa002d6

**Review by bloodyowl — APPROVED** [review]

**Pull request merged**